ci/ci/ci.py

```diff
@@ -206,6 +206,31 @@
 gh_router = gh_routing.Router()
 
 
+@routes.post('/invalidate_and_retest')
+@check_csrf_token
+@web_authenticated_developers_only
+async def invalidate_and_retest(request):
+    app = request.app
+    dbpool = app['dbpool']
+    batch_client = app['batch_client']
+    post = await request.post()
+    branch_id = post['watched_branch'].strip()
+    pr_number = post['pull_request_number'].strip()
```

Comment:
Why do you need both pr_number and branch_id? Isn't the pr_number sufficient?

Reply:
CI can support multiple tracked branches, right?